### PR TITLE
Create Reusable Loading Component and implement it

### LINKED ISSUE #853: Create Reusable Loading Component and implement it
## Description

Currently, loading states are handled inconsistently across different components in our application. This leads to code duplication and potential inconsistencies in user experience. We need a centralized approach to manage loading states. 

## Current Behavior

Loading states are implemented separately in various components, often using different methods (e.g., skeletons, spinners) and styles.

## Proposed Solution

Create a reusable `LoadingWrapper` component that can be used across the application to standardize loading UI and behavior.

### Example Implementation

```typescript
import React from 'react';
import { Skeleton, CircularProgress } from '@mui/material';

type LoadingType = 'skeleton' | 'spinner';

interface LoadingWrapperProps {
  isLoading: boolean;
  loadingType?: LoadingType;
  children: React.ReactNode;
  width?: string | number;
  height?: string | number;
}

const LoadingWrapper: React.FC<LoadingWrapperProps> = ({
  isLoading,
  loadingType = 'skeleton',
  children,
  width = '100%',
  height = 100
}) => {
  if (!isLoading) return <>{children}</>;

  switch (loadingType) {
    case 'skeleton':
      return <Skeleton variant="rectangular" width={width} height={height} />;
    case 'spinner':
      return <CircularProgress />;
    default:
      return null;
  }
};

export default LoadingWrapper;
```

This component can be used to wrap content that might be in a loading state:

```typescript
<LoadingWrapper isLoading={isLoading} loadingType="skeleton">
  <YourComponent />
</LoadingWrapper>
```

By implementing this `LoadingWrapper`, we can standardize loading states across the application, improving consistency and reducing code duplication. To test your component please implement it once on `/quests/page.tsx` by replacing our current code here 

```tsx
        {loading ? (
          <FeaturedQuestSkeleton />
        ) : (
 ```

### PR DESCRIPTION
**Pull request by Ugo-X**
Resolves: #853 

- Feature

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `Loading` component to enhance loading state representation with customizable indicators (skeleton or spinner).
	- Improved user experience by maintaining redirection to the "/admin" route if the token expires.
	- Quests are now fetched automatically when the user's account address changes.

- **Bug Fixes**
	- Ensured proper handling of user token expiry and quest fetching logic.

### CODE REVIEW

File: app/admin/quests/page.tsx [hunk 1]
```diff
@@ -91,7 +92,9 @@ export default function Page() {
           </div>
         </div>
         {loading ? (
-          <FeaturedQuestSkeleton />
+          <Loading isLoading={loading} loadingType="skeleton">
+            <FeaturedQuestSkeleton />
+          </Loading>
```
Comment (fricoben) [inline app/admin/quests/page.tsx]: What's the point of creating <Loading /> to do that ? 

Here you need to wrap the section directly and delete the ternary with the loading like so:
```tsx
          <Loading isLoading={loading} loadingType="skeleton">
            <section className={homePagestyles.section}>
            ....
```

**Review by fricoben — CHANGES_REQUESTED** [review]
I don't think you understand why we created the wrapper please refactor

**Comment by Ugo-X** [comment]
> I don't think you understand why we created the wrapper please refactor

I actually thought i did that already, guess i skipped while rushing

File: app/admin/quests/page.tsx [hunk 2]
```diff
@@ -17,7 +17,8 @@ import homePagestyles from "@styles/Home.module.css";
 import { a11yProps } from "@components/UI/tabs/a11y";
 import { CustomTabPanel } from "@components/UI/tabs/customTab";
 import { Tab, Tabs } from "@mui/material";
-import FeaturedQuestSkeleton from "@components/skeletons/questsSkeleton";
+// import FeaturedQuestSkeleton from "@components/skeletons/questsSkeleton";
```
Comment (fricoben) [inline app/admin/quests/page.tsx]: Delete comment and delete component now that we don't use it anymore

**Review by fricoben — CHANGES_REQUESTED** [review]
1. After testing on `/admin/quests` it doesn't work please fix
2. Delete component and comment now that we don't use it anymore

**Comment by Ugo-X** [comment]
@fricoben Gm

**Review by fricoben — APPROVED** [review]
lgtm

**Pull request merged**